DataSynapseApi/ArmoniK.DevelopmentKit.GridServer.Client/Service.cs

```diff
@@ -249,6 +267,67 @@ public string Submit(string methodName, byte[] argument, IServiceInvocationHandl
                     handler);
     }
 
+    private byte[] ActiveGetResult(params string[] taskId)
+    {
+      return ActiveGetResults(taskId).Single().Item2;
+    }
+
+    private IEnumerable<Tuple<string, byte[]>> ActiveGetResults(IEnumerable<string> taskIds)
+    {
+      var ids      = taskIds.ToList();
+      var missing  = ids;
+      var results  = new List<Tuple<string, byte[]>>();
+      var holdPrev = 0;
+      var waitInSeconds = new List<int>
+      {
+        10,
+        1000,
+        5000,
+        10000,
+        20000,
+      };
+      var idx = 0;
+
+      Logger.BeginPropertyScope(("SessionId", SessionId), ("Function", "ActiveGetResults"));
+
+      while (missing.Count != 0)
+      {
+        var buckets = missing.Batch(10000).ToList();
```

Comment:
Do you really need to call `.ToList()` if you only iterate over it?

Reply:
IEnumerable doesn't have forEach under this object buckets. We need to transform object as a list

Reply:
`.ToList()` does useless memory allocation and must avoided. You can use the language construct `foreach` instead.
https://stackoverflow.com/a/101313